src/wallet/rpcwallet.cpp

```diff
@@ -1731,23 +1755,55 @@ UniValue listsinceblock(const JSONRPCRequest& request)
         filter = filter | ISMINE_WATCH_ONLY;
     }
 
+    bool include_removed = (request.params.size() < 4 || request.params[3].get_bool());
+
     int depth = pindex ? (1 + chainActive.Height() - pindex->nHeight) : -1;
 
     UniValue transactions(UniValue::VARR);
+    std::set<uint256> listed;
 
     for (map<uint256, CWalletTx>::iterator it = pwalletMain->mapWallet.begin(); it != pwalletMain->mapWallet.end(); it++)
     {
         CWalletTx tx = (*it).second;
 
         if (depth == -1 || tx.GetDepthInMainChain() < depth)
-            ListTransactions(tx, "*", 0, true, transactions, filter);
+        {
+            if (ListTransactions(tx, "*", 0, true, transactions, filter) && include_removed)
+            {
+                listed.insert(it->first);
+            }
+        }
+    }
+
+    // when a reorg'd block is requested, we also list any relevant transactions
+    // in the blocks of the chain that was detached
+    UniValue removed(UniValue::VARR);
+    while (include_removed && paltindex && paltindex != pindex)
+    {
+        CBlock block;
+        if (!ReadBlockFromDisk(block, paltindex, Params().GetConsensus()))
+        {
+            throw JSONRPCError(RPC_INTERNAL_ERROR, "Can't read block from disk");
```

Comment:
A 0-conf tx will be in the mempool right? Presuming it is, it will in fact be in the transactions array.

The allow partial parameter makes sense to me. I think I'll do that for starters until we make a decision (which may be another PR).